### DISCUSSION TITLE
Directional lights meet Wolfram Language behavior

### DISCUSSION THREAD

**Discussion by TiagoCavalcante** [Mathics3/mathics-threejs-backend#69] [Ideas]
Directional lights in mathics-threejs-backend have been using [spherical coordinates](https://en.wikipedia.org/wiki/Spherical_coordinate_system) since its creation, but Wolfram Language doesn't do this way, it uses cartesian coordinates + ImageScaled coordinates for things like `Lighting -> Automatic`.

I think we should implement ImageScaled and change directional lights to cartesian coordinates.

@axkr that is why lighting in Symja doesn't look the same way as it does in Wolfram Cloud.

**Comment by TiagoCavalcante** [comment]
@rocky your thoughts?

**Reply by rocky** [reply]
@TiagoCavalcante I am always in favor of keeping behavior as compatible with Mathematica. 

In some situations like adding better SymPy interaction we may extend things, but keeping the behavior the same is just a good thing to do.

If it wasn't done in the past in my opinion it was probably more because it wasn't thought of or because it was too difficult to do. So if you think we can do it now, we should.

That's my view.

**Reply by TiagoCavalcante** [reply]
@rocky I believe directional lights were made this way because then we can "unrotate" it by applying some transformations with the camera rotation. The other way to do this is using custom shaders (and this is way faster to rotate the camera with multiple directional lights). It is basically O(n) vs O(0).

**Reply by rocky** [reply]
Ok - thanks for the explanation. Onward and upward!
